tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -389,3 +392,133 @@ def finalize(self, state):
+  moments as a function of lower ones, even if not requested, all lower
+  moments will be computed as well. Note, while any arbitrarily high
+  central moment is theoretically supported, `RunningCentralMoments` cannot
+  guarantee numerical stability for all moments.
+
+  `RunningCentralMoments` objects do not hold state information. That
+  information, which includes intermediate calculations, are held in a
+  `RunningCentralMomentsState` as returned via `initialize` and `update`
+  method calls.
+
+  #### References
+  [1]: Philippe Pebay. Formulas for Robust, One-Pass Parallel Computation of
+        Covariances and Arbitrary-Order Statistical Moments. _Technical Report
+        SAND2008-6212_, 2008.
+        https://prod-ng.sandia.gov/techlib-noauth/access-control.cgi/2008/086212.pdf
+  """
+
+  def __init__(self, shape, max_moment, dtype=tf.float32):
+    """Instantiates this object.
+
+    Args:
+      shape: Python `Tuple` or `TensorShape` representing the shape of
+        incoming samples.
+      max_moment: Integer that represents the highest central moment of
+        interest.
+      dtype: Dtype of incoming samples and the resulting statistics.
+        By default, the dtype is `tf.float32`. Any integer dtypes will be
+        cast to corresponding floats (i.e. `tf.int32` will be cast to
+        `tf.float32`), as intermediate calculations should be performing
+        floating-point division.
+    """
+    self.shape = shape
+    self.max_moment = max_moment
+    if dtype is tf.int64:
+      dtype = tf.float64
+    elif dtype.is_integer:
+      dtype = tf.float32
+    self.dtype = dtype
+    self.mean_stream = RunningMean(
+        self.shape, self.dtype
+    )
+
+  def initialize(self):
+    """Initializes an empty `RunningCentralMomentsState`.
+
+    Returns:
+      state: `RunningCentralMomentsState` representing a stream of no
+        inputs.
+    """
+    return RunningCentralMomentsState(
+        mean_state=self.mean_stream.initialize(),
+        sum_exponentiated_residuals=tf.zeros(
+            (self.max_moment,) + self.shape, self.dtype),
+    )
+
+  def update(self, state, new_sample):
+    """Update the `RunningCentralMomentsState` with a new sample.
+
+    Args:
+      state: `RunningCentralMomentsState` that represents the current
+        state of running statistics.
+      new_sample: Incoming `Tensor` sample with shape and dtype compatible with
+        those used to form the `RunningCentralMomentsState`.
+
+    Returns:
+      state: `RunningCentralMomentsState` with updated calculations.
+    """
+    n_2 = 1
+    n_1 = state.mean_state.num_samples
+    n = tf.cast(n_1 + n_2, dtype=self.dtype)
+    delta_mean = new_sample - state.mean_state.mean
+    new_mean_state = self.mean_stream.update(state.mean_state, new_sample)
+    old_res = state.sum_exponentiated_residuals
+    new_sum_exponentiated_residuals = [tf.zeros(self.shape, self.dtype)]
```

Comment:
These zeros don't seem to ever be adjusted.  Is there a reason to store them in the output, since they are always zero?

Reply:
What ends up happening is the `new_sum_exponentiated_residuals` gets converted to a tensor with a leading dimension that indexes into the (i+1)th central moment (i.e. the 0th index is the first central moment). Since the first central moment is always 0 I prepend this list with a corresponding `tf.zeros` to simplify the `Tensor` of moments that's returned.